public/app/features/explore/state/query.ts

```diff
@@ -696,6 +698,114 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
+    const interpolateCorrelationHelperVars =
+      isCorrelationEditorMode && !isLeftPane && correlationEditorHelperData !== undefined;
+    let newQuerySource: Observable<ExplorePanelData>;
+
+    // Filter queries by those explicitly requested by refId
+    const logQueries = queryResponse.logsResult?.queries || [];
+    const queries = logQueries.map((query) => ({
+      ...query,
+      datasource: query.datasource || datasourceInstance?.getRef(),
+    }));
+
+    if (!hasNonEmptyQuery(queries) || !datasourceInstance) {
+      return;
+    }
+
+    // Some datasource's query builders allow per-query interval limits,
+    // but we're using the datasource interval limit for now
+    const minInterval = datasourceInstance?.interval;
+
+    const queryOptions: QueryOptions = {
+      minInterval,
+      // maxDataPoints is used in:
+      // Loki - used for logs streaming for buffer size, with undefined it falls back to datasource config if it supports that.
+      // Elastic - limits the number of datapoints for the counts query and for logs it has hardcoded limit.
+      // Influx - used to correctly display logs in graph
+      // TODO:unification
+      // maxDataPoints: mode === ExploreMode.Logs && datasourceId === 'loki' ? undefined : containerWidth,
+      maxDataPoints: containerWidth,
+    };
+
+    let scopedVars: ScopedVars = {};
+    if (interpolateCorrelationHelperVars && correlationEditorHelperData !== undefined) {
+      Object.entries(correlationEditorHelperData?.vars).forEach((variable) => {
+        scopedVars[variable[0]] = { value: variable[1] };
+      });
+    }
+
+    const timeZone = getTimeZone(getState().user);
+    const range = getTimeRange(
+      timeZone,
+      {
+        from: dateTimeForTimeZone(timeZone, absoluteRange.from),
+        to: dateTimeForTimeZone(timeZone, absoluteRange.to),
+      },
+      getFiscalYearStartMonth(getState().user)
+    );
+    const transaction = buildQueryTransaction(exploreId, queries, queryOptions, range, false, timeZone, scopedVars);
+
+    dispatch(changeLoadingStateAction({ exploreId, loadingState: LoadingState.Loading }));
+
+    newQuerySource = combineLatest([runRequest(datasourceInstance, transaction.request), correlations$]).pipe(
+      mergeMap(([data, correlations]) =>
+        decorateData(
+          // Query splitting, otherwise duplicates results
+          data.state === LoadingState.Done ? mergeDataSeries(queryResponse, data) : data,
+          queryResponse,
+          absoluteRange,
+          undefined,
+          queries,
+          correlations,
+          showCorrelationEditorLinks,
+          defaultCorrelationEditorDatasource
+        )
+      )
+    );
+
+    newQuerySource.subscribe({
+      next(data) {
+        if (data.logsResult !== null && data.state === LoadingState.Done) {
+          reportInteraction('grafana_explore_logs_result_displayed', {
```

Comment:
Would it be worth to add some param here to distinguish between running a query and reloading more results? Otherwise it may just increase `grafana_explore_logs_result_displayed` stats without adding more context.

Reply:
Absolutely! Done.